src/multi_vector_simulator/server.py

```diff
@@ -107,13 +118,33 @@ def run_simulation(json_dict, epa_format=True, **kwargs):
     logging.debug("Accessing script: B0_data_input_json")
     dict_values = B0.convert_from_json_to_special_types(json_dict)
 
+    # if True will return the lp file's content in dict_values
+    lp_file_output = dict_values[SIMULATION_SETTINGS][OUTPUT_LP_FILE][VALUE]
+    # to avoid the lp file being saved somewhere on the server
+    dict_values[SIMULATION_SETTINGS][OUTPUT_LP_FILE][VALUE] = False
+
     print("")
     logging.debug("Accessing script: C0_data_processing")
     C0.all(dict_values)
 
     print("")
     logging.debug("Accessing script: D0_modelling_and_optimization")
-    results_meta, results_main = D0.run_oemof(dict_values)
+    results_meta, results_main, local_energy_system = D0.run_oemof(
+        dict_values, return_les=True
+    )
+
+    if lp_file_output is True:
+        logging.debug("Saving the content of the model's lp file")
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            local_energy_system.write(
+                os.path.join(tmpdirname, "lp_file.lp"),
+                io_options={"symbolic_solver_labels": True},
+            )
+            with open(os.path.join(tmpdirname, "lp_file.lp")) as fp:
+                file_content = fp.read()
+
+        dict_values[SIMULATION_SETTINGS][OUTPUT_LP_FILE][VALUE] = file_content
+        dict_values[SIMULATION_SETTINGS][OUTPUT_LP_FILE][UNIT] = TYPE_STR
```

Comment:
I am not sure if this is a duplication with the existing code.

Reply:
It is duplication but at a use level of the modules (`server.py:run_simulation` is like a main file making use of the different modules for the specific use of mvs on a server)